fabtests/multinode/src/core_coll.c

```diff
@@ -119,15 +121,14 @@ static int wait_for_comp(void *ctx)
 	return err;
 }
 
-static int coll_setup()
+static int coll_setup_w_start_addr_stride(int start_addr, int stride)
 {
 	int err;
-	struct fi_av_set_attr av_set_attr;
 
-	av_set_attr.count = pm_job.num_ranks;
-	av_set_attr.start_addr = 0;
+	av_set_attr.count = 0;
+	av_set_attr.start_addr = start_addr;
 	av_set_attr.end_addr = pm_job.num_ranks - 1;
-	av_set_attr.stride = 1;
+	av_set_attr.stride = stride;
```

Comment:
We probably need to verify that we have at least 2 ranks that will participate in the operation, or skip the test.  E.g.
```
if (start_addr + stride > end_addr)
    return -FI_ENOSYS;
```

Reply:
The is_my_rank_participating() test will make the test a no-op if no ranks are participating (-n1) and works when only one rank is participating in the collective (-n 2 with start_addr=1)